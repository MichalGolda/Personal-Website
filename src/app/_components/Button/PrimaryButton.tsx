import Button from "./Button";
import type { ButtonProps } from "./Button";

const PrimaryButton: React.FC<React.PropsWithChildren<ButtonProps>> = ({
  className = "",
  children,
  ...props
}) => {
  return (
    <Button
      className={`bg-primary text-white hover:bg-primary/90 active:bg-primary/80 border border-transparent border-solid ${className}`}
      {...props}
    >
      {children}
    </Button>
  );
};

export default PrimaryButton;
