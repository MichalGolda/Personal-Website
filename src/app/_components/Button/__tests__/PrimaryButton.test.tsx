import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import PrimaryButton from "../PrimaryButton"

describe("PrimaryButton component", () => {
    it("renders without errors", () => {
        render(<PrimaryButton type="button">Button test</PrimaryButton>)
    
        const button = screen.getByRole("button")

        expect(button).toBeInTheDocument()
        expect(button).toHaveAttribute("type", "button")
        expect(button).toHaveClass("bg-primary", "text-white", "hover:bg-primary/90", "active:bg-primary/80", "border-transparent")
    })

    it("renders unchanged", () => {
        const { container } = render(<PrimaryButton type="button">Button test</PrimaryButton>)

        expect(container).toMatchSnapshot()
    })
})