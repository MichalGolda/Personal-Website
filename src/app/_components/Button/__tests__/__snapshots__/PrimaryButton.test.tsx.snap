// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton component renders unchanged 1`] = `
<div>
  <button
    class="py-4 px-8 font-bold text-sm md:text-base transition-all duration-500 disabled:bg-lightGrey disabled:text-body bg-primary text-white hover:bg-primary/90 active:bg-primary/80 border border-transparent border-solid "
    type="button"
  >
    Button test
  </button>
</div>
`;
